### PR TITLE
feat: update braze sdk to latest version 6.6.0

### PR DESCRIPTION
**Pull request by 1abhishekpandey**
# Description

- Updated the Braze SDK to the latest version `6.6.0`.
- Changed the Pod from `Appboy-iOS-SDK` to `BrazeKit`.
- The Push Notification API has undergone a reimplementation.
- Updated Readme.
- Updated outdated pod dependency task

### CODE REVIEW

File: Rudder-Braze/Classes/RudderBrazeIntegration.m [hunk 1]
```diff
@@ -449,6 +383,47 @@ - (ConnectionMode)getConnectionMode:(NSDictionary *)config {
...
+-(void)didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
+    [self->braze.notifications registerDeviceToken:deviceToken];
+    [RSLogger logInfo:@"Braze registerDeviceToken:"];
+}
+
+// - Add support for silent notification
+
+- (void)didReceiveRemoteNotification:(NSDictionary *)userInfo fetchCompletionHandler:(void (^)(UIBackgroundFetchResult))completionHandler {
+    // As asked in the Braze doc: https://www.braze.com/docs/developer_guide/platform_integration_guides/swift/push_notifications/integration/#step-3-enable-push-handling, push integration code is called in application’s main thread.
+    dispatch_async(dispatch_get_main_queue(), ^{
```
Comment (pallabmaiti) [inline Rudder-Braze/Classes/RudderBrazeIntegration.m]: We should not use thread. We should forward it to the API. It is more flexibility to the developer if they want to use the thread or not.!

File: Rudder-Braze/Classes/RudderBrazeIntegration.m [hunk 2]
```diff
@@ -449,6 +383,47 @@ - (ConnectionMode)getConnectionMode:(NSDictionary *)config {
...
+
+-(void)didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
+    [self->braze.notifications registerDeviceToken:deviceToken];
+    [RSLogger logInfo:@"Braze registerDeviceToken:"];
+}
+
+// - Add support for silent notification
+
+- (void)didReceiveRemoteNotification:(NSDictionary *)userInfo fetchCompletionHandler:(void (^)(UIBackgroundFetchResult))completionHandler {
+    // As asked in the Braze doc: https://www.braze.com/docs/developer_guide/platform_integration_guides/swift/push_notifications/integration/#step-3-enable-push-handling, push integration code is called in application’s main thread.
```
Comment (pallabmaiti) [inline Rudder-Braze/Classes/RudderBrazeIntegration.m]: Add this to the header file.

**Review by pallabmaiti — APPROVED** [review]

File: Rudder-Braze/Classes/RudderBrazeIntegration.m (hunk 1 above)
Comment (1abhishekpandey) [reply Rudder-Braze/Classes/RudderBrazeIntegration.m]: I just wanted to understand/confirm, that even though it is mentioned to call those Push APIs in the main thread, should we pass on the responsibility to the developer?

Comment (pallabmaiti) [reply Rudder-Braze/Classes/RudderBrazeIntegration.m]: Yes. If we are implementing the thread, then we are restricting the use of the thread. A developer may or may not want to use it.

**Review by pallabmaiti — APPROVED** [review]

**Review by desusai7 — APPROVED** [review]

**Comment by 1abhishekpandey** [comment]
Closing this PR, as release [PR](https://github.com/rudderlabs/rudder-integration-braze-ios/pull/25) is raised.

**Pull request closed without merging**